application/teams/team.py

```diff
@@ -1,14 +1,15 @@
 """Access to Team objects in Ne4J"""
+from uuid import uuid4
 from neomodel import StructuredNode, StringProperty, RelationshipTo, cardinality
 
 from application.model_mixins import DefaultPropertyMixin, DefaultHelperMixin
 from application.teams.team_schema import TeamSchema
 
-
 class Team(StructuredNode, DefaultPropertyMixin, DefaultHelperMixin):  # pylint:disable=abstract-method
     """The class that manages Teams"""
     schema = TeamSchema
     name = StringProperty(required=True)
+    join_key = StringProperty(required=False, default=uuid4)
```

Comment:
Should the join key be a uuid? I think it could be a lot shorter?

Reply:
UUID was decided on after internal discussion.

Reply:
I'd prefer a shorter join link.
This could lead to potential collisions (but still very unlikely just like uuids). But it makes the links more human friendly and if we enable regeneration later on collision wont be a huge problem.
You can use this for that:
```
shortId = RandomStringUtils.randomAlphanumeric(8);
```
https://stackoverflow.com/questions/4267475/generating-8-character-only-uuids as reference